cmd/kubeadm/app/cmd/init.go

```diff
@@ -269,6 +270,11 @@ func (i *Init) Run(out io.Writer) error {
 		return fmt.Errorf("couldn't parse kubernetes version %q: %v", i.cfg.KubernetesVersion, err)
 	}
 
+	// Upload currently used configuration to the cluster
+	if err := saveconfigphase.SaveConfiguration(i.cfg, client); err != nil {
```

Comment:
I have a bit of a nitpick about the verb here. "Save" is a little vague to me, and immediately brings up questions like "what happens if the config already exists? Will it get clobbered? Will it just save a new copy? Is it expected that this configuration was already created, and I'm just saving an update to it?"

I like how our other APIs are extremely informative about their contracts, like `tokenphase.CreateBootstrapConfigMapIfNotExists()` or `tokenphase.UpdateOrCreateToken()`. Can the phase be named something like `configphase` and the method be `CreateOrUpdateConfiguration()`? That also gives us the flexibility of also having other complementary APIs in the same phase like `CreateConfigIfNotExists()` in the future if there's demand for other variants.